src/api/channel.rs

```diff
@@ -0,0 +1,34 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::clock::ClockInt;
+use embedded_time::duration::Milliseconds;
+
+pub enum SendOrRecvStatus {
+    Timeout,
+    Sent,
+    Received,
+}
+
+pub struct SendOrRecvError;
+
+pub type SendOrRecvResult = Result<SendOrRecvStatus, SendOrRecvError>;
+
+pub trait CtapHidChannel {
```

Comment:
I'd say it's more readable if the trait has the same name as the module. The names could be `hid_channel` and `HidChannel` respectively for module and trait. I don't think we need the CTAP. But I think we need the HID (unless channel is an HID-specific concept). @kaczmarczyck what do you think? Is channel only HID-related? Or is it possible to have a non-HID channel?

Reply:
Since it takes a `[u8; 64]`, it is indeed HID specific. For other naming issues, see my other comment. The reason I suggested "CtapHid" is because it implements a way to send CTAPHID packets. But since we are in a CTAP context, we can drop that if you like.